src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System.Private.Runtime.InteropServices.JavaScript.Tests.csproj

```diff
@@ -18,13 +22,30 @@
     <Compile Include="System\Runtime\InteropServices\JavaScript\Http\HttpRequestMessageTest.cs" />
     <Compile Include="System\Runtime\InteropServices\JavaScript\ParallelTests.cs" />
     <Compile Include="System\Runtime\InteropServices\JavaScript\TimerTests.cs" />
+    <Compile Include="System\Runtime\InteropServices\JavaScript\UriMarshaller.cs" />
+    <None Include="$(BaseIntermediateOutputPath)\GeneratedFiles\JavaScript.MarshalerGenerator\System.Runtime.InteropServices.JavaScript.JsImportGenerator\JsImport.g.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="invokeJs.js" />
+    <None Include="timers.js" />
   </ItemGroup>
 
   <ItemGroup>
     <WasmExtraFilesToDeploy Include="timers.js" />
+    <WasmExtraFilesToDeploy Include="invokeJs.js" />
   </ItemGroup>
   <ItemGroup>
     <!-- Part of the shared framework but not exposed. -->
     <ProjectReference Include="..\src\System.Private.Runtime.InteropServices.JavaScript.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\gen\MarshalerGenerator\JavaScript.MarshalerGenerator.csproj" OutputItemType="Analyzer" ReferenceOutputAssembly="false" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <!-- Use following lines to write the generated files to disk. -->
+    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles>
+    <CompilerGeneratedFilesOutputPath>$(BaseIntermediateOutputPath)\GeneratedFiles</CompilerGeneratedFilesOutputPath>
+  </PropertyGroup>
```

Comment:
nit: move to the top to the other project properties. Also technically you would want to use `IntermediateOutputPath` instead of `BaseIntermediateOutputPath` as the output is configuration and tfm dependent. What makes this tricky is that the `IntermediateOutputPath` property isn't accessible outside of a targets file or outside of a target in the project file as it gets evaluated later by the SDK.